src/dash-playlist-loader.js

```diff
@@ -295,6 +243,111 @@ export default class DashPlaylistLoader extends EventTarget {
     }
   }
 
+  requestErrored_(err, request, startingState) {
```

Comment:
triggers a request error if it finds an error and returns true if there was an error so request functions can exit early. We used this logic in many places in the original code and we use it in two places now.